src/main/thrift/parquet.thrift

```diff
@@ -883,13 +928,42 @@ struct ColumnChunk {
 
   /** Encrypted column metadata for this chunk **/
   9: optional binary encrypted_column_metadata
+  /** 
+    * The column order for this chunk.
+    * 
+    * If not set readers should check FileMetadata.column_orders
+    * instead.
+    *
+    * Populated in both PAR1 and PAR3
+    */
+  10: optional ColumnOrder column_order
+  /** Set to true if all pages in the column chunk are dictionary 
+    * encoded 
+    */
+  11: optional bool all_pages_dictionary_encoded
+ /** 
+   * The index to the SchemaElement in FileMetadata for this 
+   * column.
+   */
+  12: optional i32 schema_index 
```

Comment:
what use case is there for having column chunks be out of order? Wouldn't it be easier to just specify that the column chunks have to be in the same order as the schema?

Reply:
Also note that without the order being the same as the schema, the whole "random access" idea of this PR goes out of the window.

It doesn't help me to have a MetadataPage with random access, if I don't know which column I need to access in the first place.

Instead, what I want is that if I want to access the third column in the schema, then I need to access the third column chunk.

Reply:
As discussed on the mailing list, how about this alternative design:

How about just turning things around: Instead of having a schema_index in the ColumnMetadata, we could have a column_metadata_index in the schema. If that index is missing/-1, then this signifies that the column is empty, so no metadata will be present for it. With this, we would get the best of both worlds: We would always have O(1) random I/O even in case of such empty columns (as we would use the column_metadata_index for the lookup) and we would not need to store any ColumnMetadata for empty columns.

After given this a second thought, this also makes more sense in general. As the navigation direction is usually always from schema to metadata (not vice versa!), the schema should point us to the correct metadata instead of the metadata pointing us to the correct schema entry.

Reply:
@JFinis this would only be useful for columns which are empty throughout all row groups, right? As opposed to skipping _some_ column chunks but not all.